kb(grid):added kb on how to pre-set checkbox filter on initial load (#587)
telerik/blazor-docs: knowledge-base/grid-pre-select-checkbox-filter.md

---

> title: How can I pre-set a checkbox filter using a filter descriptor?

Reviewer: "can i" is unlikely to be used in a Google search.

```suggestion
title: How to pre-set a checkbox filter with a filter descriptor?
```

description: 
type: how-to

> page_title: How can I pre-set a checkbox filter using a filter descriptor?

Reviewer: ```suggestion
page_title: How to pre-set a checkbox list filter using a filter descriptor?
```

slug: grid-pre-select-checkbox-filter
position: 
tags: grid,checkbox,filter
ticketid: 1541980
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Grid for Blazor</td>
		</tr>
	</tbody>
</table>

## Description
How can I pre-set a checkbox filter using a filter descriptor? How to set checkbox filter on Grid first load? How can I statically specify the checkbox list content? How to set checkbox filter on the first initialization of the Grid?

> ## Solution

Reviewer: The solution is missing several steps - 
* creating a new GridState instance
* adding the new CompositeFilterDescriptor instance to the `FilterDescriptors` collection in the state
* applying the new state to the current Grid state

Author: Done.

> To pre-select checkbox filter in Grid using filter descriptor:

Reviewer: ```suggestion
To pre-select checkbox filter in the Grid using a filter descriptor:
```

1. Use the [OnStateInit](https://docs.telerik.com/blazor-ui/components/grid/state#events) event of the Grid.

> 2. Implement **CompositeFilterDescriptor** inside the OnStateInit handler.

Reviewer: ```suggestion
2. Implement a **CompositeFilterDescriptor** inside the OnStateInit handler.
```
